### ISSUE TITLE
Update implementation to use AppTree

### ISSUE THREAD

**Issue by bwesner** [adamsoffer/next-apollo#51]
In your `next-apollo-example`, you mention waiting on a bug fix to be able to use `useRouter` properly: https://github.com/adamsoffer/next-apollo-example/blob/52317222c6a326568056ffa1ed63793fa423d10d/components/Header/index.js#L7

They have fixed it in https://github.com/zeit/next.js/pull/7732

We need to update `next-apollo` to use `AppTree` so that this can be properly supported. Afterwords, `next-apollo-example` should be updated.

Check out how they updated their example here: https://github.com/zeit/next.js/commit/dc9e2708984bc6bc7e87724fb5193e7f788c89a3

**Comment by adamsoffer** [comment]
Great thanks for the heads up @bwesner.

**Comment by adamsoffer** [comment]
@bwesner I'm testing with next `v9.0.5-canary.1` but this is still an issue for me. Getting the following error:

```
GraphQL error occurred [getDataFromTree] TypeError: Cannot read property 'query' of null
```

**Comment by adamsoffer** [comment]
@bwesner Sorry been quite busy and haven't gotten to this yet. Feel free to submit a PR!

**Comment by adamsoffer** [comment]
@bwesner heads up this has been fixed in `v3.1.4`
